### DISCUSSION TITLE
I am looking for method to open notifications in iOS appium python client.

### DISCUSSION THREAD

**Discussion by poobalanc** [appium/appium#16162] [Q&A]
I am new to iOS automation and I am looking for method to open notifications in iOS appium python client.
Any help is very much appreciated..

Thanks,
Poobalan.

**Comment by RSM-SBA** [comment]
(edit : sorry i have this only for java not for python)
i think simply open it does not work.

but maybe its enough for you to check if the Notification is thrown

https://github.com/appium/appium-xcuitest-driver#mobile-expectnotification

Or you do not initilize the appium driver for the app itself (because you have than not the possibility to access the System via appium) and instead with the springboard. This way can lead to high efforts to keep it running, when iOS changing things with iOS updates.
So i would not use it excessivly

**capabilities for driver setup :**
`caps.setCapability("app", "com.apple.springboard");
`

**install your app manually not via appium setup:**
Map<String, Object> params2 = new HashMap<>();
params2.put("app", "PATH_TO_YOUR_APP");
driver.executeScript("mobile: installApp", params2);

**you can start/switch apps than :**

This for starting your app (evetually with arguments if needed)
final List<String> processArgs = new ArrayList<>(Arrays.asList("your_app_arguments"));
		HashMap<String, Object> args = new HashMap<>();
		args.put("bundleId", "your_app_bundle_id");
		args.put("arguments", processArgs);
		driver.executeScript("mobile: launchApp", args);

**starting springboard app to access system apps (like notifications)**
driver.activateApp("com.apple.springboard");

**Reply by poobalanc** [reply]
Thank you so much Stephan,
I am very new to appium iOS, i am not sure if above code supported in python? will give a try and let you know..

**Reply by RSM-SBA** [reply]
similar functions should be available in the python client, but i dont have the syntax in mind. Maybe Google can help you.
If you just starting with your test project, maybe you switch to java. Personally i found more helpful sources regarding appium for java.

Additionally appium.pro is good source for beginners. (mostly in java).
There is also an article testing ios push notifications

https://appiumpro.com/editions/15-testing-ios-push-notifications.
